compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -3492,6 +3494,10 @@ protected ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLo
             case CHECKING_EXPRESSION_END:
                 endContext();
                 return ParserRuleContext.EXPRESSION_RHS;
+            case ON_FAIL_CHECK_END:
+                endContext();
+                endContext();
```

Comment:
Since we close to contexts, this comment will help understand what ctxs we are ending
```suggestion
                endContext(); // end on-fail-check
                endContext(); // end checking-expr
```